hatch-dependency-coversion/src/hatch_dependency_coversion/metadata_hook.py

```diff
@@ -38,8 +41,23 @@ def _update_dependency_versions(
 
     def update(self, metadata: dict[str, Any]) -> None:
         """Update metadata for coversioning."""
+        # this is from https://github.com/flying-sheep/hatch-docstring-description/blob/main/src/hatch_docstring_description/read_description.py
+        # and would prevent surprise recursions, and if that author is worried about it then so am I
+        if ("update", __file__) in (
+            (frame.function, frame.filename) for frame in inspect.stack()[1:]
+        ):
+            return
```

Comment:
Could you elaborate on how this recursion would actually happen?

Reply:
Nope! Like I said, I just saw this in another plugin and thought "better safe than sorry"